Terraform crashes during blueprint deployment

When performing a terraform apply for a plan that deploys an existing blueprint, terraform crashes. The error shown in the command window is attached below, and the crash.log file is attached.

cdombrowski-a01:bin cdombrowski$ terraform apply
data.vra_project.this: Refreshing state...
data.vra_cloud_account_vsphere.this: Refreshing state...

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # vra_deployment.this will be created
  + resource "vra_deployment" "this" {
      + blueprint_id         = "9b0d6fd9-2520-46a0-9927-5f49750555a3"
      + blueprint_version    = "4"
      + catalog_item_id      = (known after apply)
      + catalog_item_version = (known after apply)
      + created_at           = (known after apply)
      + created_by           = (known after apply)
      + description          = "terraform test deployment"
      + expand_last_request  = true
      + expand_resources     = true
      + expense              = (known after apply)
      + id                   = (known after apply)
      + inputs               = {
          + "inCMI"     = "Cloud-init"
          + "inCount"   = "3"
          + "inFlavor"  = "c5.large"
          + "inLicense" = "rhat"
          + "inRail"    = "protected"
          + "inTenant"  = "A"
          + "vmName"    = "chrisd_terraform"
        }
      + last_updated_at      = (known after apply)
      + last_updated_by      = (known after apply)
      + lease_expire_at      = (known after apply)
      + name                 = "chrisd_terraform"
      + project              = (known after apply)
      + project_id           = "79b572c5-92f4-4cbc-bacc-d9e7b7b09f84"
      + resources            = (known after apply)
      + simulated            = (known after apply)
      + status               = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

vra_deployment.this: Creating...
vra_deployment.this: Still creating... [10s elapsed]
vra_deployment.this: Still creating... [20s elapsed]
vra_deployment.this: Still creating... [30s elapsed]
vra_deployment.this: Still creating... [40s elapsed]
vra_deployment.this: Still creating... [50s elapsed]
vra_deployment.this: Still creating... [1m0s elapsed]
vra_deployment.this: Still creating... [1m10s elapsed]
vra_deployment.this: Still creating... [1m20s elapsed]
vra_deployment.this: Still creating... [1m30s elapsed]
vra_deployment.this: Still creating... [1m40s elapsed]
vra_deployment.this: Still creating... [1m50s elapsed]
vra_deployment.this: Still creating... [2m0s elapsed]
vra_deployment.this: Still creating... [2m10s elapsed]

Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is json.Number, not string
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: goroutine 12 [running]:
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: github.com/vmware/terraform-provider-vra/vra.expandInputs(0x1db28a0, 0xc000582450, 0x7)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/github/vmware/terraform-provider-vra/vra/structure.go:148 +0x175
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: github.com/vmware/terraform-provider-vra/vra.resourceDeploymentRead(0xc0004bcb60, 0x1cf1ac0, 0xc0005ac8c0, 0x1, 0x1)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/github/vmware/terraform-provider-vra/vra/resource_deployment.go:353 +0xe78
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: github.com/vmware/terraform-provider-vra/vra.resourceDeploymentCreate(0xc0004bcb60, 0x1cf1ac0, 0xc0005ac8c0, 0x2, 0x2cf66c0)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/github/vmware/terraform-provider-vra/vra/resource_deployment.go:276 +0xb7e
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc000441380, 0xc0006523c0, 0xc0005acbe0, 0x1cf1ac0, 0xc0005ac8c0, 0x1, 0xc0007c8c08, 0x1db04a0)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.0/helper/schema/resource.go:305 +0x3b4
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc0005fe500, 0xc00064fa68, 0xc0006523c0, 0xc0005acbe0, 0xc0000cef28, 0xc0000be3b8, 0x1db28a0)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.0/helper/schema/provider.go:294 +0x18f
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00000e008, 0x2244f60, 0xc0001b6210, 0xc00023a120, 0xc00000e008, 0xc0001b6210, 0xc0007b3bd0)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.0/internal/helper/plugin/grpc_provider.go:885 +0x884
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1f48f00, 0xc00000e008, 0x2244f60, 0xc0001b6210, 0xc00023a0c0, 0x0, 0x2244f60, 0xc0001b6210, 0xc000168000, 0x4fa)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.1.0/internal/tfplugin5/tfplugin5.pb.go:3189 +0x23e
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000d82c0, 0x2252e40, 0xc000558d80, 0xc000140000, 0xc00029b0b0, 0x2cc4e80, 0x0, 0x0, 0x0)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:995 +0x466
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: google.golang.org/grpc.(*Server).handleStream(0xc0000d82c0, 0x2252e40, 0xc000558d80, 0xc000140000, 0x0)
2019-11-11T11:04:09.481-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1275 +0xda6
2019-11-11T11:04:09.482-0500 [DEBUG] plugin.terraform-provider-vra: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0002b4040, 0xc0000d82c0, 0x2252e40, 0xc000558d80, 0xc000140000)
2019-11-11T11:04:09.482-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:710 +0x9f
2019-11-11T11:04:09.482-0500 [DEBUG] plugin.terraform-provider-vra: created by google.golang.org/grpc.(*Server).serveStreams.func1
2019-11-11T11:04:09.482-0500 [DEBUG] plugin.terraform-provider-vra: 	/Users/markpeek/go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:708 +0xa1
2019/11/11 11:04:09 [DEBUG] vra_deployment.this: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2019/11/11 11:04:09 [TRACE] EvalMaybeTainted: vra_deployment.this encountered an error during creation, so it is now marked as tainted
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalWriteState
2019/11/11 11:04:09 [TRACE] EvalWriteState: removing state object for vra_deployment.this
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalApplyProvisioners
2019/11/11 11:04:09 [TRACE] EvalApplyProvisioners: vra_deployment.this has no state, so skipping provisioners
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2019/11/11 11:04:09 [TRACE] EvalMaybeTainted: vra_deployment.this encountered an error during creation, so it is now marked as tainted
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalWriteState
2019/11/11 11:04:09 [TRACE] EvalWriteState: removing state object for vra_deployment.this
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalIf
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalIf
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalWriteDiff
2019/11/11 11:04:09 [TRACE] <root>: eval: *terraform.EvalApplyPost
2019/11/11 11:04:09 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: rpc error: code = Unavailable desc = transport is closing
2019/11/11 11:04:09 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2019/11/11 11:04:09 [TRACE] [walkApply] Exiting eval tree: vra_deployment.this
2019/11/11 11:04:09 [TRACE] vertex "vra_deployment.this": visit complete
2019/11/11 11:04:09 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2019-11-11T11:04:09.484-0500 [DEBUG] plugin: plugin process exited: path=/Users/cdombrowski/terraform/bin/terraform-provider-vra pid=7278 error="exit status 2"
2019/11/11 11:04:09 [TRACE] dag/walk: upstream of "provider.vra (close)" errored, so skipping
2019/11/11 11:04:09 [TRACE] dag/walk: upstream of "root" errored, so skipping
2019/11/11 11:04:09 [TRACE] statemgr.Filesystem: creating backup snapshot at terraform.tfstate.backup
2019/11/11 11:04:09 [TRACE] statemgr.Filesystem: state has changed since last snapshot, so incrementing serial to 108
2019/11/11 11:04:09 [TRACE] statemgr.Filesystem: writing snapshot at terraform.tfstate
2019/11/11 11:04:09 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2019/11/11 11:04:09 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2019-11-11T11:04:09.498-0500 [DEBUG] plugin: plugin exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

[crash.log](https://github.com/vmware/terraform-provider-vra/files/3832201/crash.log)

--- a/examples/deployment/blueprint/README.md
+++ b/examples/deployment/blueprint/README.md
@@ -10,7 +10,7 @@ There are variables which need to be added to terraform.tfvars.
 * `refresh_token` - The refresh token for the vRA account
 * `project_name` - Project Name
 * `blueprint_id` - Blueprint ID
-* `blueprint_version` - Blueprint Version
+* `blueprint_version` - Blueprint Version. Leave it empty for deployments with current draft of the blueprint.
 * `deployment_name` - Deployment Name
 
 To facilitate adding these variables, a sample tfvars file can be copied first:

--- a/examples/deployment/blueprint/main.tf
+++ b/examples/deployment/blueprint/main.tf
@@ -16,8 +16,10 @@ resource "vra_deployment" "this" {
   project_id        = data.vra_project.this.id
 
   inputs = {
-    flavor   = "small"
-    image    = "centos"
+    flavor = "small"
+    image  = "centos"
+    count  = 2
+    flag   = true
   }
 
   expand_resources    = true

--- a/vra/structure.go
+++ b/vra/structure.go
@@ -139,13 +139,12 @@ func flattenAndNormalizeCLoudAccountGcpRegionIds(regionOrder []string, cloudAcco
 	return m, nil
 }
 
-// expandInputs will convert the interface  into a map of strings
-func expandInputs(configInputs interface{}) map[string]string {
-	inputs := make(map[string]string)
-
+// expandInputs will convert the interface  into a map of interface
+func expandInputs(configInputs interface{}) map[string]interface{} {
+	inputs := make(map[string]interface{})
 	for key, value := range configInputs.(map[string]interface{}) {
 		if value != nil {
-			inputs[key] = value.(string)
+			inputs[key] = fmt.Sprint(value)
 		}
 	}
 

--- a/vra/structure_test.go
+++ b/vra/structure_test.go
@@ -1,0 +1,44 @@
+package vra
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestExpandInputs(t *testing.T) {
+	type Message struct {
+		Name string
+		Body string
+		Time int64
+	}
+	m := Message{"Foo", "Bar", 1294706395881547000}
+
+	inputs := make(map[string]interface{})
+	inputs["whole"] = 2
+	inputs["text"] = "This is string"
+	inputs["flag"] = true
+	inputs["fraction"] = 2.4
+	inputs["message"] = m
+
+	expandedInputs := expandInputs(inputs)
+
+	if expandedInputs["whole"] != fmt.Sprint(2) {
+		t.Errorf("int type input is not expanded correctly.")
+	}
+
+	if expandedInputs["text"] != "This is string" {
+		t.Errorf("string type input is not expanded correctly.")
+	}
+
+	if expandedInputs["flag"] != fmt.Sprint(true) {
+		t.Errorf("bool type input is not expanded correctly.")
+	}
+
+	if expandedInputs["fraction"] != fmt.Sprint("2.4") {
+		t.Errorf("float type input is not expanded correctly.")
+	}
+
+	if expandedInputs["message"] != fmt.Sprint(m) {
+		t.Errorf("float type input is not expanded correctly.")
+	}
+}

